impl/graphsync_test.go

```diff
@@ -982,6 +982,8 @@ func TestNetworkDisconnect(t *testing.T) {
 	tracing := collectTracing(t)
 	traceStrings := tracing.TracesToStrings()
 	require.Contains(t, traceStrings, "response(0)->executeTask(0)")
+	require.Contains(t, traceStrings, "response(0)->abortRequest(0)")
+	require.NotContains(t, traceStrings, "response(0)->abortRequest(1)")
 	// may contain multiple abortRequest traces as the network error can bubble up >1 times
 	// but these will only record if the request is still executing
 	require.Contains(t, traceStrings, "request(0)->newRequest(0)")
```

Comment:
Can we be precise in this test now (most others are precise in exactly what traces we expect, the `Contains` ones are lose to allow for some variation, which is unfortunate), e.g.:

```suggestion
	require.ElementsMatch(t, []string{
		"response(0)->executeTask(0)",
		"response(0)->abortRequest(1)",
		"request(0)->newRequest(0)",
		"request(0)->executeTask(0)",
		"request(0)->terminateRequest(0)",
	}, tracing.TracesToStrings())
```

Reply:
I think so.  Lemme get reliable tests and then I will merge